### PR TITLE
Trigger webhooks

### PR DESCRIPTION
**Pull request by attiks**

### CODE REVIEW

**Comment by attiks** [comment]
See https://github.com/UN-OCHA/assessmentregistry8-site/pull/198

**Comment by attiks** [comment]
Still WIP, and also decide what to do for bulk commands

File: html/modules/custom/docstore/src/Controller/DocumentController.php [hunk 1]
```diff
@@ -568,6 +568,9 @@ protected function createDocumentFromParameters(NodeType $node_type, array $para
     // Invalidate cache.
     Cache::invalidateTags(['documents']);
 
+    // Trigger webhook.
+    docstore_notify_webhooks($node_type->id() . ':create', $this->prepareEntityResourceDataForResponse($document, $provider, TRUE));
```
Comment (orakili) [inline html/modules/custom/docstore/src/Controller/DocumentController.php:572]: I think that should be moved to the `createDocument()` function to be consistent with the `update` and `delete` notifications.

Also that would be triggered for every document in bulk request, which may not be what we want.

**Review by orakili — COMMENTED** [review]
@attiks I don't think we should return the full data because that doesn't handle the "private" state of the resource.

I think we should limit to the `uuid` and `resource type` and let the subscribers query the API to get the full data so that `private` content is properly handled.

For that, I would propose to have the following events (ex for `create` on **documents** but would be similar for `update`, `delete` and **terms**) that basically return only the `uuid` and resource type.

- `document:create` (fired for any document type)
  - data: `{"uuid': "document uuid", "type': "node type id"}`
- `document:{bundle}:create` (fired for documents of type {bundle})
  - data: `"document uuid"`
  
 Both would fired in the createDocument():
 
 ```php
 docstore_notify_webhooks('document:' . $node_type->id() . ':create', $data['uuid']);
 docstore_notify_webhooks('document:create', [
   'uuid' => $data['uuid'],
   'type' => $node_type->id(),
 ]);
```

Same thing for the `updated` and `delete` actions and also for the **terms**:

 ```php
 docstore_notify_webhooks('term:' . $vocabulary->id() . ':create', $data['uuid']);
 docstore_notify_webhooks('term:create', [
   'uuid' => $data['uuid'],
   'vocabulary' => $vocabulary->id(),
 ]);
```

This way a provider could subscribe to all notifications on documents/terms or to specific resource types.

Creating a PR with the proposal.

**Comment by attiks** [comment]
> Creating a PR with the proposal.

Feel free to update this one

File: html/modules/custom/docstore/src/Controller/DocumentController.php (hunk 1 above)
Comment (attiks) [reply html/modules/custom/docstore/src/Controller/DocumentController.php:572]: Good catch, should not be fired for bulk operations.

**Comment by orakili** [comment]
> Feel free to update this one

Ok, finishing some work on tests and I'll update the PR.

**Comment by orakili** [comment]
Created another PR because I didn't feel like solving merge conflicts: https://github.com/UN-OCHA/docstore-site/pull/124

**Pull request closed without merging**